pulsar-client-cpp/include/pulsar/BoostHash.h

```diff
@@ -0,0 +1,39 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef BOOST_HASH_HPP_
+#define BOOST_HASH_HPP_
+
+#include <pulsar/Hash.h>
+
+#include <cstdint>
+#include <string>
+#include <boost/functional/hash.hpp>
```

Comment:
I think this class should go in the `lib/` directory since it's should not be part of the user API. As a user, I just need to know about the "Boost" hash enumerator and not its implementation